blaze/server/server.py

```diff
@@ -406,3 +406,42 @@ def compserver():
         'data': result,
         'names': expr.fields
     })
+
+@api.route('/add', methods=['POST', 'HEAD', 'OPTIONS'])
+@crossdomain(origin='*', methods=['POST', 'HEAD', 'OPTIONS'])
+@authorization
+def addserver():
+    content_type = request.headers['content-type']
+    matched = mimetype_regex.match(content_type)
+
+    if matched is None:
+        return 'Unsupported serialization format %s' % content_type, 415
+
+    try:
+        serial = _get_format(matched.groups()[0])
+    except KeyError:
+        return (
+            "Unsupported serialization format '%s'" % matched.groups()[0],
+            415,
+        )
+
+    try:
+        payload = serial.loads(request.data)
+    except ValueError:
+        return ("Bad data.  Got %s " % request.data, 400)  # 400: Bad Request
+
+
+    data = _get_data.cache[flask.current_app]
+
+    data_not_mm_msg = ("Cannot update blaze server data since its current data"
+                       " is a %s and not a mutable mapping (dictionary like).")
+    if not isinstance(data, collections.MutableMapping):
+        return (data_not_mm_msg % type(data), 400)
```

Comment:
maybe a `422 Unprocessable Entity` would be better here since the syntax of the message is okay